modules/sdk-coin-sol/src/lib/instructionParamsFactory.ts

```diff
@@ -367,29 +362,35 @@ const ataInitInstructionKeysIndexes = {
  */
 function parseAtaInitInstructions(instructions: TransactionInstruction[]): Array<AtaInit | Memo> {
   const instructionData: Array<AtaInit | Memo> = [];
-  const ataInitInstruction = instructions[ataInitInstructionIndexes.InitializeAssociatedTokenAccount];
-
-  const mintAddress = ataInitInstruction.keys[ataInitInstructionKeysIndexes.MintAddress].pubkey.toString();
-  const tokenName = findTokenName(mintAddress);
 
-  const ataInit: AtaInit = {
-    type: InstructionBuilderTypes.CreateAssociatedTokenAccount,
-    params: {
-      mintAddress,
-      ataAddress: ataInitInstruction.keys[ataInitInstructionKeysIndexes.ATAAddress].pubkey.toString(),
-      ownerAddress: ataInitInstruction.keys[ataInitInstructionKeysIndexes.OwnerAddress].pubkey.toString(),
-      payerAddress: ataInitInstruction.keys[ataInitInstructionKeysIndexes.PayerAddress].pubkey.toString(),
-      tokenName,
-    },
-  };
-  instructionData.push(ataInit);
+  for (const instruction of instructions) {
+    const type = getInstructionType(instruction);
+    switch (type) {
+      case ValidInstructionTypesEnum.Memo:
+        const memo: Memo = { type: InstructionBuilderTypes.Memo, params: { memo: instruction.data.toString() } };
+        instructionData.push(memo);
+        break;
```

Comment:
this _might_ break things that used to depend on the mint instruction being the last instruction.

lets try to preserve this behavior.
can do something like:
```
let memo?: Memo

...

case ValidInstructionTypesEnum.Memo:
  memo = { type: InstructionBuilderTypes.Memo, params: { memo: instruction.data.toString() } };

...

// after all instructions
if (memo) {
   instructions.push(memo);
}
```